crates/worker/src/commands/compile.rs

```diff
@@ -44,8 +40,24 @@ pub async fn do_compile(
     let hardhat_config_path = workspace_path.join("hardhat.config.ts");
 
     // instantly create the directories
-    tokio::fs::create_dir_all(&workspace_path).await?;
-    tokio::fs::create_dir_all(&artifacts_path).await?;
+    tokio::fs::create_dir_all(&workspace_path)
+        .await
+        .map_err(anyhow::Error::from)
+        .with_context(|| {
+            format!(
+                "Couldn't create workspace dir: {}",
+                workspace_path.display()
+            )
+        })?;
+    tokio::fs::create_dir_all(&artifacts_path)
+        .await
+        .map_err(anyhow::Error::from)
```

Comment:
Why do we use the `anyhow::Error::from`, can't we just implement `From<...>` for our `CompilationError`?

Reply:
Noticed that can be with_context, it wraps to anyhow automatically.
with regards to:

> Why do we use the anyhow::Error::from, can't we just implement From<...> for our CompilationError?

Not done this way since while every anyhow::Error is CompilationError::UnknownError, not every io::Error is CompilationError::UnknownError